/**
 * Copyright 2020 Huawei Technologies Co., Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package ge.proto;

enum DataType
{
  DT_UNDEFINED = 0;  // Used to indicate a DataType field has not been set.
  DT_FLOAT     = 1;  // float type
  DT_FLOAT16   = 2;  // fp16 type
  DT_INT8      = 3;  // int8 type
  DT_UINT8     = 4;  // uint8 type
  DT_INT16     = 5;  // int16 type
  DT_UINT16    = 6;  // uint16 type
  DT_INT32     = 7;  //
  DT_INT64     = 8;  // int64 type
  DT_UINT32    = 9;  // unsigned int32
  DT_UINT64    = 10;  // unsigned int64
  DT_BOOL      = 11;  // bool type
  DT_DOUBLE    = 12; // double type
  DT_STRING = 13;            // string type
  DT_DUAL_SUB_INT8 = 14;    /**< dual output int8 type */
  DT_DUAL_SUB_UINT8 = 15;    /**< dual output uint8 type */
  DT_COMPLEX64 = 16;         // complex64 type
  DT_COMPLEX128 = 17;        // complex128 type
  DT_QINT8 = 18;             // qint8 type
  DT_QINT16 = 19;            // qint16 type
  DT_QINT32 = 20;            // qint32 type
  DT_QUINT8 = 21;            // quint8 type
  DT_QUINT16 = 22;           // quint16 type
  DT_RESOURCE  = 23;         // resource type
  DT_STRING_REF = 24;        // string_ref type
  DT_DUAL      = 25;              /**< dual output type */
}